README.md

```diff
@@ -10,3 +10,79 @@ Includes:
+
+# A Quick Intro to CNBs:
+### All of the sample buildpacks in this repo implement the [CNB Spec](https://github.com/buildpack/spec/blob/master/buildpack.md)
+
+## Detect vs Build
+Each buildpack should have two executables files, `detect` and `build`, which get run during the detect phase and the build phase. The [lifecycle](https://github.com/buildpack/lifecycle) will run the `detect` files for all available buildpacks (parallelized by default), and then run all of the `build` files (serially) for buildpacks which succesfully passed the detect phase.
+
+## Buildplan 
+The buildplan is a `toml` file that buildpacks can write to. It is the medium through which data is passed from the `detect` phase to the `build` phase. 
+
+The form of an entry is 
+```
+
+[<dependency name>]
+version = "<dependency version>"
+
+[<dependency name>.metadata]
+# buildpack-specific data (aka any additional metadata you need)
+
+```
+This data should then be consumed by your `build` script. For more on the consumption patterns see the [build section](#build) below.
+
+## Detect
+This phase will run the `bin/detect` executable in the buildpack
+
+Determine whether your buildpack is needed for the given application, (e.g: look for a `package.json`, if you are providing `node_modules`) and do one of the following:
+  - Write a(n) entry(ies) to the `buildplan`, and return a zero exit status.
+  - Fail (without writing to the `buildplan`), and return a non-zero exit status. 
+
+## Build
+This phase will run the `bin/build` executable in the buildpack
+
+Read the buildplan, and determine if this buildpack is needed. If so, contribute to the final application image. This contribution could be as simple as setting some environment variables within the image, creating a `layer` containing a binary (e.g: `node`, `python`, or `ruby`), or adding app dependencies (e.g: running `npm install`, `pip install -r requirements.txt`, or `bundle install`).
+
+## Layers
+These are folders in the image, where your dependencies live. They have the general form: `/layers/buildpack.id/<layer_name>`. For example, the `node_modules` layer contributed by the `cloudfoundry/npm-cnb` would be `/layers/org.cloudfoundry.npm/node_modules`
+
+
+## Buildpack vs builder
+A **builder** contains internal operating system images, the lifecycle executables for your application, and a collection of Cloud Native Buildpacks, all structured according to the [CNB specification](https://github.com/buildpack/spec). Builders are [created by](https://github.com/buildpack/pack#working-with-builders-using-create-builder), and [consumed by](https://github.com/buildpack/pack#example-building-using-the-default-builder-image), the [pack CLI](https://github.com/buildpack/pack). 
```

Comment:
Similar to above, the pack repo now points to buildpacks.io, rather than duplicating the info. You should update these links, which will be dead otherwise.